language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/package.scala

```diff
@@ -15,15 +15,17 @@ import com.squareup.javapoet._
 
 import scala.collection.JavaConverters._
 
+package inner {
+  case class FieldInfo(damlName: String, damlType: Type, javaName: String, javaType: TypeName)
+}
+
```

Comment:
> what exactly does it buy us?

A class or object in a package object must be accessed by reference to that package object. If it is in a package, you can access it directly. This difference is mostly transparent to Scala code, but not to the JVM, or Java code (where that matters).

Additionally, because package objects have been historically buggy, it is best to minimize what you put in them.

> why public type alieas `type Fields = IndexedSeq[FieldInfo]` is not part of the `package inner`?

This is not allowed.

Reply:
By the way, in Scala, the Java-style `package` declaration[s] at top of file are [just syntactic sugar](https://www.artima.com/pins1ed/packages-and-imports.html#13.1) for the full `package x { ... }` form; the `{ }` are implicitly placed around the rest of the file.